test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -43,6 +49,25 @@
               (println "timed-out waiting for events")
               @dest)))))))
 
+(defn wait-for-exactly-event
+  "Waits for event to land in dest. On first registered event or events in dest,
+  determines whether the registered event(s) match the supplied expected
+  event. If so, returns the expected event. Otherwise returns contents of dest."
+  [dest expected-event]
+  (let [start-time (System/currentTimeMillis)]
+    (loop []
```

Comment:
So I could do this but `elapsed-time` of the recur condition `(< elapsed-time wait-time)` is dynamically redefined/rebound every iteration which appears harder with `while` and we're also waiting for the atom value to be populated.
I could do: `(while (< (- (System/currentTimeMillis) start-time) wait-time)` and avoiding binding `elapsed-time` but that seems less readable to me. It basically ends up being something like "just wait until one of these conditions is true", ie:
```
  (let [start-time (System/currentTimeMillis)]
    (while (and (empty? @dest)
                (< (- (System/currentTimeMillis) start-time) wait-time)))
    (if (empty? @dest)
      (do
        (println "timed-out waiting for events")
         @dest)
      (if (exactly-matches-event? dest expected-event)
        expected-event
        @dest)
```
this seems really weird to me so i'm sure there's a better way